### ISSUE TITLE
Null resolve context

### ISSUE THREAD

**Issue by tutushkin** [Abc-Arbitrage/Abc.MoqComplete#23]
We've got a request of the Rider user:
https://youtrack.jetbrains.com/issue/RIDER-52957

There is a problem in the plugin at https://github.com/Abc-Arbitrage/Abc.MoqComplete/blob/master/Abc.MoqComplete/Abc.MoqComplete/Services/TestProjectProvider.cs,
passing 'null' instead of proper ModuleReferenceResolveContext.

Please ping me if you need any details on how to acquire the context here.

**Comment by ltrzesniewski** [comment]
The YouTrack issue you pointed to is private. Could you copy the relevant content here?

**Comment by lconstan** [comment]
@tutushkin actually the method `GetReferences` is obsolete, the best would be to update it to `IPsiModule.GetModuleReference` as suggested.
Do you have any hints on how to get the "`toModule`" and "`ModuleReferenceResolveContext`" parameters for that case?

P.S. : A PR would be welcome :)

**Comment by tutushkin** [comment]
@ltrzesniewski sure, the user complains that thair code is red.
We've found this exception in their logs:
17:11:27.182 |E|                               | JetPool(S) #5:14   | Unknown Resolve Context

--- EXCEPTION #1/2 [ArgumentException]
Message = “Unknown Resolve Context”
ExceptionPath = Root.InnerException
ClassName = System.ArgumentException
HResult = E_INVALIDARG=COR_E_ARGUMENT=WIN32_ERROR_INVALID_PARAMETER=80070057
Source = JetBrains.Platform.ProjectModel
StackTraceString = “
  bei JetBrains.ProjectModel.Assemblies.Impl.ResolveContextManager.GetResolveResult(AssemblyNameInfo assemblyNameInfo, IModuleReferenceResolveContext context)
     bei JetBrains.ProjectModel.Assemblies.Impl.ModuleReferencesResolveStore.GetResolveResult(IModuleToAssemblyReference reference)
     bei JetBrains.ProjectModel.Assemblies.Impl.ModuleReferencesResolveStore.<>c.<GetReferencedAssemblies>b__28_0(IAssemblyToAssemblyReference reference)
     bei System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
     bei System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
     bei System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
     bei JetBrains.ReSharper.Psi.Modules.ProjectAssemblyPsiModule.GetReferences(IModuleReferenceResolveContext resolveContext)
     bei Abc.MoqComplete.Services.TestProjectProvider.IsTestProject(IPsiModule psiModule) in D:\a\Abc.MoqComplete\Abc.MoqComplete\Abc.MoqComplete\Abc.MoqComplete\Services\TestProjectProvider.cs:Zeile 23.
     bei JetBrains.ReSharper.Feature.Services.CodeCompletion.Infrastructure.ItemsProviderOfSpecificContext`1.JetBrains.ReSharper.Feature.Services.CodeCompletion.Infrastructure.ICodeCompletionItemsProvider.IsAvailable(ISpecificCodeCompletionContext context)
     bei JetBrains.Util.Logging.Logger.CatchIgnoreException[TValue,TException](Func`1 F, TValue defaultValue)
”

**Comment by tutushkin** [comment]
@lconstan if you'll decide to update, I suggest using the version without the toModule for your case.

**ModuleReferenceResolveContext** is the thing that defines which module's references to use.
But as I see in the stacktrace, **IsTestProject** is called from inside the completion and proper MRRC is already set into the cookie so you can safelly pass **null** to updated API.

tldr; **psiModule.GetPsiServices().Modules.GetModuleReferences(psiModule);** should do the trick.

**Comment by lconstan** [comment]
I made the change with the above commit. Thanks tutushkin!
